---
title: "DESeq2_RG_SDELCISAR_13"
author: "dd"
date: "2/17/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

<center><h2>**RNAseq Analysis**</h3></center><br>  



```{r Load Packages,warning=FALSE,message=FALSE}
library(DESeq2)
library(stringr)
library(pheatmap)
library(RColorBrewer)
library(ggrepel)
library(EnsDb.Hsapiens.v79)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(clusterProfiler)
library(ReactomePA)
```


```{r Input Files and Design Matrix, echo=FALSE}
###Input Count files from STAR
DESeq2DIR="/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2"
sampleSheetFILE="/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/sampleSheet.csv"
designmatFILE="/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/designmatrix.txt"
normcount_outFILE="/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/normalized_counts_log2_star.txt"
rlog_outFILE="/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/rlog_star.txt"

FDR_cutoff=0.01

###Design Matrix

sampleSheet=read.delim(sampleSheetFILE,sep=",")
samples=sampleSheet$SampleNAME
sampleTable <- data.frame(Sample = factor(samples), FileName = sampleSheet$FileNAME, Condition = factor(sampleSheet$Condition), Treatment = factor(sampleSheet$Treatment))
rownames(sampleTable)=samples
sampleTable

dd1=read.delim(designmatFILE, header=T)
dd1$Condition=relevel(as.factor(dd1$Condition),ref="3")
dd1$Treatment=relevel(as.factor(dd1$Treatment),ref="RN")
dd1$group=factor(paste0(dd1$Condition,dd1$Treatment))

design=model.matrix(~ 0+group,data=dd1)

se_star <- DESeqDataSetFromHTSeqCount(sampleTable = sampleTable,
                    directory = DESeq2DIR,
                    design = design)

```

```{r DESeq2 - Norm counts, size factors and Statistical model fit}

## Filter genes with low counts
se_star <- se_star[rowSums(counts(se_star)) > 10, ]

## Number of genes left after low-count filtering:
nrow(se_star)

#Fit Statistical model#
se_star2 <- DESeq(se_star)

##Size Factors calculated by DESeq2, raw counts are divided by these values#
sizeFactors(se_star2)

## Transform counts for visualization
se_rlog <- rlog(se_star2) # For exploratory analysis, blind=T #
rld <- assay(rlog(se_star2, blind=FALSE)) # For downstream analysis, blind=F # 

```

<center><h2>**Exploratory Plots**</h3></center><br>  



```{r Exploratory Plots, echo=FALSE}
## Build Heatmap

# calculate between-sample distance matrix
sampleDistMatrix <- as.matrix(dist(t(assay(se_rlog))))
pheatmap(sampleDistMatrix)



## Principal component analysis

# With Individual replicates labeled
pcaData <- plotPCA(se_rlog, intgroup=c("Treatment"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=Treatment)) + geom_point(size=3) + geom_text_repel(data=pcaData,aes(label=name),size=3) + xlab(paste0("PC1: ",percentVar[1],"% variance")) + ylab(paste0("PC2: ",percentVar[2],"% variance")) + theme_bw() + coord_fixed()

```



<center><h3>*Treatment RK vs. RN*</h3></center><br> 


```{r DESeq2 - DE Analysis, Treatment RK vs. RN , warning=FALSE, echo=FALSE}
## Differential expression analysis

resultsNames(se_star2)

for(timept in c(3,7,14,21))
{
treatname=paste("group",timept,"RK",sep="")  
ctname=paste("group",timept,"RN",sep="")  

de <- results(object = se_star2, contrast=list(treatname,ctname))
de
summary(de)
#Add gene names
ensembl_genenames=str_replace(rownames(de),pattern=".[0-9]+$",replacement="")
hgnc_genenames <- ensembldb::select(EnsDb.Hsapiens.v79, keys=ensembl_genenames, keytype = "GENEID", columns = c("SYMBOL","GENEID"))
de$hgnc_symbol=cbind(ensembl_genenames,hgnc_genenames[match(ensembl_genenames,hgnc_genenames$GENEID),1])[,2]


# Select genes for which padj < FDR_cutoff
de_select <- de[de$padj < FDR_cutoff & !is.na(de$padj),]
de_selectUP <- subset(de_select,log2FoldChange>0)
de_selectDOWN <- subset(de_select,log2FoldChange<0)

head(de_selectUP[order(de_selectUP$padj),])
head(de_selectDOWN[order(de_selectDOWN$padj),])


#Enrichment Analysis
## Run GO enrichment analysis##
#UP genes#
assign(paste("egoUP_",timept,"RK",sep=""),enrichGO(gene = str_replace(rownames(de_selectUP),pattern=".[0-9]+$",replacement=""), universe = str_replace(rownames(de),pattern=".[0-9]+$",replacement=""), keyType = "ENSEMBL", OrgDb = org.Hs.eg.db, ont = "ALL", pAdjustMethod = "BH", qvalueCutoff = 0.05, readable = TRUE))

tt1=data.frame(get(paste("egoUP_",timept,"RK",sep="")))
write.table(tt1, file=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/GO/egoUP_",timept,"RK.txt",sep=""), sep="\t", quote=F, row.names = F)

#DOWN genes#
assign(paste("egoDOWN_",timept,"RK",sep=""),enrichGO(gene = str_replace(rownames(de_selectDOWN),pattern=".[0-9]+$",replacement=""), universe = str_replace(rownames(de),pattern=".[0-9]+$",replacement=""), keyType = "ENSEMBL", OrgDb = org.Hs.eg.db, ont = "ALL", pAdjustMethod = "BH", qvalueCutoff = 0.05, readable = TRUE))

tt1=data.frame(get(paste("egoDOWN_",timept,"RK",sep="")))
write.table(tt1, file=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/GO/egoDOWN_",timept,"RK.txt",sep=""), sep="\t", quote=F, row.names = F)

##Pathway enrichment analysis##
#UP genes#
entrez_ids = mapIds(org.Hs.eg.db,
                    keys=str_replace(rownames(de_selectUP),pattern=".[0-9]+$",replacement=""), 
                    column="ENTREZID",
                    keytype="ENSEMBL",
                    multiVals="first")
entrez_ids = entrez_ids[!is.na(entrez_ids)]
assign(paste("xx_",timept,"RK",sep=""),enrichPathway(gene=entrez_ids, pvalueCutoff = 0.05, readable=TRUE))

tt1=data.frame(get(paste("xx_",timept,"RK",sep="")))
write.table(tt1, file=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/GO/ReactomeUP_",timept,"RK.txt",sep=""), sep="\t", quote=F, row.names = F)

#DOWN genes#
entrez_ids = mapIds(org.Hs.eg.db,
                    keys=str_replace(rownames(de_selectDOWN),pattern=".[0-9]+$",replacement=""), 
                    column="ENTREZID",
                    keytype="ENSEMBL",
                    multiVals="first")
entrez_ids = entrez_ids[!is.na(entrez_ids)]
assign(paste("yy_",timept,"RK",sep=""),enrichPathway(gene=entrez_ids, pvalueCutoff = 0.05, readable=TRUE))

tt1=data.frame(get(paste("yy_",timept,"RK",sep="")))
write.table(tt1, file=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/GO/ReactomeDOWN_",timept,"RK.txt",sep=""), sep="\t", quote=F, row.names = F)

}

```
```{r Plots, Treatment RK vs. RN Heatmaps, warning=FALSE, echo=FALSE}

Treatment_rnaseq="RK"
#T3
       
timept=3
inputFILE=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/deShrink.group",timept,Treatment_rnaseq,"-group",timept,"RN.txt",sep="")

input_data=read.delim(inputFILE,header=T)

if(timept==3 || timept==7)
{
timept2=paste("0",timept,sep="")
} else
{
timept2=timept       
}        
        
#Get rlog values #
ctrl_cols=c(paste("RN",timept2,"01",sep=""),paste("RN",timept2,"02",sep=""),paste("RN",timept2,"03",sep=""))
trt_cols=c(paste(Treatment_rnaseq,timept2,"01",sep=""),paste(Treatment_rnaseq,timept2,"02",sep=""),paste(Treatment_rnaseq,timept2,"03",sep=""))
subset_rnaseq_data=subset(rld,rownames(rld) %in% rownames(input_data),select=c(ctrl_cols,trt_cols))

pheatmap(subset_rnaseq_data, scale="row", clustering_distance_rows="correlation", clustering_distance_cols = "correlation", show_rownames =F, show_colnames =T, cluster_cols=F, fontsize_row=4, main=paste(Treatment_rnaseq,"-",timept,"-DEGenes",sep="" ))


#T7
timept=7
inputFILE=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/deShrink.group",timept,Treatment_rnaseq,"-group",timept,"RN.txt",sep="")

input_data=read.delim(inputFILE,header=T)

if(timept==3 || timept==7)
{
timept2=paste("0",timept,sep="")
} else
{
timept2=timept       
}        
        
#Get rlog values #
ctrl_cols=c(paste("RN",timept2,"01",sep=""),paste("RN",timept2,"02",sep=""),paste("RN",timept2,"03",sep=""))
trt_cols=c(paste(Treatment_rnaseq,timept2,"01",sep=""),paste(Treatment_rnaseq,timept2,"02",sep=""),paste(Treatment_rnaseq,timept2,"03",sep=""))
subset_rnaseq_data=subset(rld,rownames(rld) %in% rownames(input_data),select=c(ctrl_cols,trt_cols))

pheatmap(subset_rnaseq_data, scale="row", clustering_distance_rows="correlation", clustering_distance_cols = "correlation", show_rownames =F, show_colnames =T, cluster_cols=F, fontsize_row=4, main=paste(Treatment_rnaseq,"-",timept,"-DEGenes",sep="" ))


#T14
timept=14
inputFILE=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/deShrink.group",timept,Treatment_rnaseq,"-group",timept,"RN.txt",sep="")

input_data=read.delim(inputFILE,header=T)

if(timept==3 || timept==7)
{
timept2=paste("0",timept,sep="")
} else
{
timept2=timept       
}        
        
#Get rlog values #
ctrl_cols=c(paste("RN",timept2,"01",sep=""),paste("RN",timept2,"02",sep=""),paste("RN",timept2,"03",sep=""))
trt_cols=c(paste(Treatment_rnaseq,timept2,"01",sep=""),paste(Treatment_rnaseq,timept2,"02",sep=""),paste(Treatment_rnaseq,timept2,"03",sep=""))
subset_rnaseq_data=subset(rld,rownames(rld) %in% rownames(input_data),select=c(ctrl_cols,trt_cols))

pheatmap(subset_rnaseq_data, scale="row", clustering_distance_rows="correlation", clustering_distance_cols = "correlation", show_rownames =F, show_colnames =T, cluster_cols=F, fontsize_row=4, main=paste(Treatment_rnaseq,"-",timept,"-DEGenes",sep="" ))


#T21
timept=21
inputFILE=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/deShrink.group",timept,Treatment_rnaseq,"-group",timept,"RN.txt",sep="")

input_data=read.delim(inputFILE,header=T)

if(timept==3 || timept==7)
{
timept2=paste("0",timept,sep="")
} else
{
timept2=timept       
}        
        
#Get rlog values #
ctrl_cols=c(paste("RN",timept2,"01",sep=""),paste("RN",timept2,"02",sep=""),paste("RN",timept2,"03",sep=""))
trt_cols=c(paste(Treatment_rnaseq,timept2,"01",sep=""),paste(Treatment_rnaseq,timept2,"02",sep=""),paste(Treatment_rnaseq,timept2,"03",sep=""))
subset_rnaseq_data=subset(rld,rownames(rld) %in% rownames(input_data),select=c(ctrl_cols,trt_cols))

pheatmap(subset_rnaseq_data, scale="row", clustering_distance_rows="correlation", clustering_distance_cols = "correlation", show_rownames =F, show_colnames =T, cluster_cols=F, fontsize_row=4, main=paste(Treatment_rnaseq,"-",timept,"-DEGenes",sep="" ))

```

```{r Plots, Treatment RK vs. RN , warning=FALSE, echo=FALSE}

timept=3
treatname=paste("group",timept,"RK",sep="")  
ctname=paste("group",timept,"RN",sep="") 

dotplot(egoUP_3RK, showCategory=20, font.size=8, title=paste("Gene Ontology, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

dotplot(egoDOWN_3RK, showCategory=20, font.size=8, title=paste("Gene Ontology, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))


if(nrow(xx_3RK)>0)
{        
barplot(xx_3RK,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}
if(nrow(yy_3RK)>0)
{        
barplot(yy_3RK,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}

timept=7
treatname=paste("group",timept,"RK",sep="")  
ctname=paste("group",timept,"RN",sep="") 

dotplot(egoUP_7RK, showCategory=20, font.size=8, title=paste("Gene Ontology, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

dotplot(egoDOWN_7RK, showCategory=20, font.size=8, title=paste("Gene Ontology, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

if(nrow(xx_7RK)>0)
{        
barplot(xx_7RK,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}
if(nrow(yy_7RK)>0)
{        
barplot(yy_7RK,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}


timept=14
treatname=paste("group",timept,"RK",sep="")  
ctname=paste("group",timept,"RN",sep="") 

dotplot(egoUP_14RK, showCategory=20, font.size=8, title=paste("Gene Ontology, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

dotplot(egoDOWN_14RK, showCategory=20, font.size=8, title=paste("Gene Ontology, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))


if(nrow(xx_14RK)>0)
{        
barplot(xx_14RK,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}
if(nrow(yy_14RK)>0)
{        
barplot(yy_14RK,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}




timept=21
treatname=paste("group",timept,"RK",sep="")  
ctname=paste("group",timept,"RN",sep="") 

dotplot(egoUP_21RK, showCategory=20, font.size=8, title=paste("Gene Ontology, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

dotplot(egoDOWN_21RK, showCategory=20, font.size=8, title=paste("Gene Ontology, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))


if(nrow(xx_21RK)>0)
{        
barplot(xx_21RK,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}
if(nrow(yy_21RK)>0)
{        
barplot(yy_21RK,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}

```



<center><h3>*Treatment R6 vs. RN*</h3></center><br> 


```{r DESeq2 - DE Analysis, Treatment R6 vs. RN , warning=FALSE, echo=FALSE}
## Differential expression analysis

resultsNames(se_star2)

for(timept in c(3,7,14,21))
{
treatname=paste("group",timept,"R6",sep="")  
ctname=paste("group",timept,"RN",sep="")  

de <- results(object = se_star2, contrast=list(treatname,ctname))
de
summary(de)
#Add gene names
ensembl_genenames=str_replace(rownames(de),pattern=".[0-9]+$",replacement="")
hgnc_genenames <- ensembldb::select(EnsDb.Hsapiens.v79, keys=ensembl_genenames, keytype = "GENEID", columns = c("SYMBOL","GENEID"))
de$hgnc_symbol=cbind(ensembl_genenames,hgnc_genenames[match(ensembl_genenames,hgnc_genenames$GENEID),1])[,2]


# Select genes for which padj < FDR_cutoff
de_select <- de[de$padj < FDR_cutoff & !is.na(de$padj),]
de_selectUP <- subset(de_select,log2FoldChange>0)
de_selectDOWN <- subset(de_select,log2FoldChange<0)

head(de_selectUP[order(de_selectUP$padj),])
head(de_selectDOWN[order(de_selectDOWN$padj),])



#Enrichment Analysis
## Run GO enrichment analysis#
#UP genes#
assign(paste("egoUP_",timept,"R6",sep=""),enrichGO(gene = str_replace(rownames(de_selectUP),pattern=".[0-9]+$",replacement=""), universe = str_replace(rownames(de),pattern=".[0-9]+$",replacement=""), keyType = "ENSEMBL", OrgDb = org.Hs.eg.db, ont = "ALL", pAdjustMethod = "BH", qvalueCutoff = 0.05, readable = TRUE))

tt1=data.frame(get(paste("egoUP_",timept,"R6",sep="")))
write.table(tt1, file=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/GO/egoUP_",timept,"R6.txt",sep=""), sep="\t", quote=F, row.names = F)

#DOWN genes#
assign(paste("egoDOWN_",timept,"R6",sep=""),enrichGO(gene = str_replace(rownames(de_selectDOWN),pattern=".[0-9]+$",replacement=""), universe = str_replace(rownames(de),pattern=".[0-9]+$",replacement=""), keyType = "ENSEMBL", OrgDb = org.Hs.eg.db, ont = "ALL", pAdjustMethod = "BH", qvalueCutoff = 0.05, readable = TRUE))

tt1=data.frame(get(paste("egoDOWN_",timept,"R6",sep="")))
write.table(tt1, file=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/GO/egoDOWN_",timept,"R6.txt",sep=""), sep="\t", quote=F, row.names = F)

##Pathway enrichment analysis##
#UP genes#
entrez_ids = mapIds(org.Hs.eg.db,
                    keys=str_replace(rownames(de_selectUP),pattern=".[0-9]+$",replacement=""), 
                    column="ENTREZID",
                    keytype="ENSEMBL",
                    multiVals="first")
entrez_ids = entrez_ids[!is.na(entrez_ids)]
assign(paste("xx_",timept,"R6",sep=""),enrichPathway(gene=entrez_ids, pvalueCutoff = 0.05, readable=TRUE))

tt1=data.frame(get(paste("xx_",timept,"R6",sep="")))
write.table(tt1, file=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/GO/ReactomeUP_",timept,"R6.txt",sep=""), sep="\t", quote=F, row.names = F)

#DOWN genes#
entrez_ids = mapIds(org.Hs.eg.db,
                    keys=str_replace(rownames(de_selectDOWN),pattern=".[0-9]+$",replacement=""), 
                    column="ENTREZID",
                    keytype="ENSEMBL",
                    multiVals="first")
entrez_ids = entrez_ids[!is.na(entrez_ids)]
assign(paste("yy_",timept,"R6",sep=""),enrichPathway(gene=entrez_ids, pvalueCutoff = 0.05, readable=TRUE))

tt1=data.frame(get(paste("yy_",timept,"R6",sep="")))
write.table(tt1, file=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/GO/ReactomeDOWN_",timept,"R6.txt",sep=""), sep="\t", quote=F, row.names = F)

}
```

```{r Plots, Treatment R6 vs. RN Heatmaps, warning=FALSE, echo=FALSE}

Treatment_rnaseq="R6"
#T3
       
timept=3
inputFILE=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/deShrink.group",timept,Treatment_rnaseq,"-group",timept,"RN.txt",sep="")

input_data=read.delim(inputFILE,header=T)

if(timept==3 || timept==7)
{
timept2=paste("0",timept,sep="")
} else
{
timept2=timept       
}        
        
#Get rlog values #
ctrl_cols=c(paste("RN",timept2,"01",sep=""),paste("RN",timept2,"02",sep=""),paste("RN",timept2,"03",sep=""))
trt_cols=c(paste(Treatment_rnaseq,timept2,"01",sep=""),paste(Treatment_rnaseq,timept2,"02",sep=""),paste(Treatment_rnaseq,timept2,"03",sep=""))
subset_rnaseq_data=subset(rld,rownames(rld) %in% rownames(input_data),select=c(ctrl_cols,trt_cols))

pheatmap(subset_rnaseq_data, scale="row", clustering_distance_rows="correlation", clustering_distance_cols = "correlation", show_rownames =F, show_colnames =T, cluster_cols=F, fontsize_row=4, main=paste(Treatment_rnaseq,"-",timept,"-DEGenes",sep="" ))


#T7
timept=7
inputFILE=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/deShrink.group",timept,Treatment_rnaseq,"-group",timept,"RN.txt",sep="")

input_data=read.delim(inputFILE,header=T)

if(timept==3 || timept==7)
{
timept2=paste("0",timept,sep="")
} else
{
timept2=timept       
}        
        
#Get rlog values #
ctrl_cols=c(paste("RN",timept2,"01",sep=""),paste("RN",timept2,"02",sep=""),paste("RN",timept2,"03",sep=""))
trt_cols=c(paste(Treatment_rnaseq,timept2,"01",sep=""),paste(Treatment_rnaseq,timept2,"02",sep=""),paste(Treatment_rnaseq,timept2,"03",sep=""))
subset_rnaseq_data=subset(rld,rownames(rld) %in% rownames(input_data),select=c(ctrl_cols,trt_cols))

pheatmap(subset_rnaseq_data, scale="row", clustering_distance_rows="correlation", clustering_distance_cols = "correlation", show_rownames =F, show_colnames =T, cluster_cols=F, fontsize_row=4, main=paste(Treatment_rnaseq,"-",timept,"-DEGenes",sep="" ))


#T14
timept=14
inputFILE=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/deShrink.group",timept,Treatment_rnaseq,"-group",timept,"RN.txt",sep="")

input_data=read.delim(inputFILE,header=T)

if(timept==3 || timept==7)
{
timept2=paste("0",timept,sep="")
} else
{
timept2=timept       
}        
        
#Get rlog values #
ctrl_cols=c(paste("RN",timept2,"01",sep=""),paste("RN",timept2,"02",sep=""),paste("RN",timept2,"03",sep=""))
trt_cols=c(paste(Treatment_rnaseq,timept2,"01",sep=""),paste(Treatment_rnaseq,timept2,"02",sep=""),paste(Treatment_rnaseq,timept2,"03",sep=""))
subset_rnaseq_data=subset(rld,rownames(rld) %in% rownames(input_data),select=c(ctrl_cols,trt_cols))

pheatmap(subset_rnaseq_data, scale="row", clustering_distance_rows="correlation", clustering_distance_cols = "correlation", show_rownames =F, show_colnames =T, cluster_cols=F, fontsize_row=4, main=paste(Treatment_rnaseq,"-",timept,"-DEGenes",sep="" ))


#T21
timept=21
inputFILE=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/deShrink.group",timept,Treatment_rnaseq,"-group",timept,"RN.txt",sep="")

input_data=read.delim(inputFILE,header=T)

if(timept==3 || timept==7)
{
timept2=paste("0",timept,sep="")
} else
{
timept2=timept       
}        
        
#Get rlog values #
ctrl_cols=c(paste("RN",timept2,"01",sep=""),paste("RN",timept2,"02",sep=""),paste("RN",timept2,"03",sep=""))
trt_cols=c(paste(Treatment_rnaseq,timept2,"01",sep=""),paste(Treatment_rnaseq,timept2,"02",sep=""),paste(Treatment_rnaseq,timept2,"03",sep=""))
subset_rnaseq_data=subset(rld,rownames(rld) %in% rownames(input_data),select=c(ctrl_cols,trt_cols))

pheatmap(subset_rnaseq_data, scale="row", clustering_distance_rows="correlation", clustering_distance_cols = "correlation", show_rownames =F, show_colnames =T, cluster_cols=F, fontsize_row=4, main=paste(Treatment_rnaseq,"-",timept,"-DEGenes",sep="" ))

```




```{r Plots, Treatment R6 vs. RN , warning=FALSE, echo=FALSE}

timept=3
treatname=paste("group",timept,"R6",sep="")  
ctname=paste("group",timept,"RN",sep="") 

dotplot(egoUP_3R6, showCategory=20, font.size=8, title=paste("Gene Ontology, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

dotplot(egoDOWN_3R6, showCategory=20, font.size=8, title=paste("Gene Ontology, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))


if(nrow(xx_3R6)>0)
{        
barplot(xx_3R6,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}
if(nrow(yy_3R6)>0)
{        
barplot(yy_3R6,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}

timept=7
treatname=paste("group",timept,"R6",sep="")  
ctname=paste("group",timept,"RN",sep="") 

dotplot(egoUP_7R6, showCategory=20, font.size=8, title=paste("Gene Ontology, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

dotplot(egoDOWN_7R6, showCategory=20, font.size=8, title=paste("Gene Ontology, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

if(nrow(xx_7R6)>0)
{        
barplot(xx_7R6,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}
if(nrow(yy_7R6)>0)
{        
barplot(yy_7R6,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}


timept=14
treatname=paste("group",timept,"R6",sep="")  
ctname=paste("group",timept,"RN",sep="") 

dotplot(egoUP_14R6, showCategory=20, font.size=8, title=paste("Gene Ontology, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

dotplot(egoDOWN_14R6, showCategory=20, font.size=8, title=paste("Gene Ontology, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))


if(nrow(xx_14R6)>0)
{        
barplot(xx_14R6,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}
if(nrow(yy_14R6)>0)
{        
barplot(yy_14R6,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}




timept=21
treatname=paste("group",timept,"R6",sep="")  
ctname=paste("group",timept,"RN",sep="") 

dotplot(egoUP_21R6, showCategory=20, font.size=8, title=paste("Gene Ontology, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

dotplot(egoDOWN_21R6, showCategory=20, font.size=8, title=paste("Gene Ontology, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))


if(nrow(xx_21R6)>0)
{        
barplot(xx_21R6,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}
if(nrow(yy_21R6)>0)
{        
barplot(yy_21R6,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}

```


<center><h3>*Treatment R1 vs. RN*</h3></center><br> 


```{r DESeq2 - DE Analysis, Treatment R1 vs. RN , warning=FALSE, echo=FALSE}
## Differential expression analysis

resultsNames(se_star2)

for(timept in c(3,7,14,21))
{
treatname=paste("group",timept,"R1",sep="")  
ctname=paste("group",timept,"RN",sep="")  

de <- results(object = se_star2, contrast=list(treatname,ctname))
de
summary(de)
#Add gene names
ensembl_genenames=str_replace(rownames(de),pattern=".[0-9]+$",replacement="")
hgnc_genenames <- ensembldb::select(EnsDb.Hsapiens.v79, keys=ensembl_genenames, keytype = "GENEID", columns = c("SYMBOL","GENEID"))
de$hgnc_symbol=cbind(ensembl_genenames,hgnc_genenames[match(ensembl_genenames,hgnc_genenames$GENEID),1])[,2]


# Select genes for which padj < FDR_cutoff
de_select <- de[de$padj < FDR_cutoff & !is.na(de$padj),]
de_selectUP <- subset(de_select,log2FoldChange>0)
de_selectDOWN <- subset(de_select,log2FoldChange<0)

head(de_selectUP[order(de_selectUP$padj),])
head(de_selectDOWN[order(de_selectDOWN$padj),])


#Enrichment Analysis
## Run GO enrichment analysis#
#UP genes#
assign(paste("egoUP_",timept,"R1",sep=""),enrichGO(gene = str_replace(rownames(de_selectUP),pattern=".[0-9]+$",replacement=""), universe = str_replace(rownames(de),pattern=".[0-9]+$",replacement=""), keyType = "ENSEMBL", OrgDb = org.Hs.eg.db, ont = "ALL", pAdjustMethod = "BH", qvalueCutoff = 0.05, readable = TRUE))

tt1=data.frame(get(paste("egoUP_",timept,"R1",sep="")))
write.table(tt1, file=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/GO/egoUP_",timept,"R1.txt",sep=""), sep="\t", quote=F, row.names = F)

#DOWN genes#
assign(paste("egoDOWN_",timept,"R1",sep=""),enrichGO(gene = str_replace(rownames(de_selectDOWN),pattern=".[0-9]+$",replacement=""), universe = str_replace(rownames(de),pattern=".[0-9]+$",replacement=""), keyType = "ENSEMBL", OrgDb = org.Hs.eg.db, ont = "ALL", pAdjustMethod = "BH", qvalueCutoff = 0.05, readable = TRUE))

tt1=data.frame(get(paste("egoDOWN_",timept,"R1",sep="")))
write.table(tt1, file=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/GO/egoDOWN_",timept,"R1.txt",sep=""), sep="\t", quote=F, row.names = F)

##Pathway enrichment analysis##
#UP genes#
entrez_ids = mapIds(org.Hs.eg.db,
                    keys=str_replace(rownames(de_selectUP),pattern=".[0-9]+$",replacement=""), 
                    column="ENTREZID",
                    keytype="ENSEMBL",
                    multiVals="first")
entrez_ids = entrez_ids[!is.na(entrez_ids)]
assign(paste("xx_",timept,"R1",sep=""),enrichPathway(gene=entrez_ids, pvalueCutoff = 0.05, readable=TRUE))

tt1=data.frame(get(paste("xx_",timept,"R1",sep="")))
write.table(tt1, file=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/GO/ReactomeUP_",timept,"R1.txt",sep=""), sep="\t", quote=F, row.names = F)

#DOWN genes#
entrez_ids = mapIds(org.Hs.eg.db,
                    keys=str_replace(rownames(de_selectDOWN),pattern=".[0-9]+$",replacement=""), 
                    column="ENTREZID",
                    keytype="ENSEMBL",
                    multiVals="first")
entrez_ids = entrez_ids[!is.na(entrez_ids)]
assign(paste("yy_",timept,"R1",sep=""),enrichPathway(gene=entrez_ids, pvalueCutoff = 0.05, readable=TRUE))

tt1=data.frame(get(paste("yy_",timept,"R1",sep="")))
write.table(tt1, file=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/GO/ReactomeDOWN_",timept,"R1.txt",sep=""), sep="\t", quote=F, row.names = F)

}

```


```{r Plots, Treatment R1 vs. RN Heatmaps, warning=FALSE, echo=FALSE}

Treatment_rnaseq="R1"
#T3
       
timept=3
inputFILE=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/deShrink.group",timept,Treatment_rnaseq,"-group",timept,"RN.txt",sep="")

input_data=read.delim(inputFILE,header=T)

if(timept==3 || timept==7)
{
timept2=paste("0",timept,sep="")
} else
{
timept2=timept       
}        
        
#Get rlog values #
ctrl_cols=c(paste("RN",timept2,"01",sep=""),paste("RN",timept2,"02",sep=""),paste("RN",timept2,"03",sep=""))
trt_cols=c(paste(Treatment_rnaseq,timept2,"01",sep=""),paste(Treatment_rnaseq,timept2,"02",sep=""),paste(Treatment_rnaseq,timept2,"03",sep=""))
subset_rnaseq_data=subset(rld,rownames(rld) %in% rownames(input_data),select=c(ctrl_cols,trt_cols))

pheatmap(subset_rnaseq_data, scale="row", clustering_distance_rows="correlation", clustering_distance_cols = "correlation", show_rownames =F, show_colnames =T, cluster_cols=F, fontsize_row=4, main=paste(Treatment_rnaseq,"-",timept,"-DEGenes",sep="" ))


#T7
timept=7
inputFILE=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/deShrink.group",timept,Treatment_rnaseq,"-group",timept,"RN.txt",sep="")

input_data=read.delim(inputFILE,header=T)

if(timept==3 || timept==7)
{
timept2=paste("0",timept,sep="")
} else
{
timept2=timept       
}        
        
#Get rlog values #
ctrl_cols=c(paste("RN",timept2,"01",sep=""),paste("RN",timept2,"02",sep=""),paste("RN",timept2,"03",sep=""))
trt_cols=c(paste(Treatment_rnaseq,timept2,"01",sep=""),paste(Treatment_rnaseq,timept2,"02",sep=""),paste(Treatment_rnaseq,timept2,"03",sep=""))
subset_rnaseq_data=subset(rld,rownames(rld) %in% rownames(input_data),select=c(ctrl_cols,trt_cols))

pheatmap(subset_rnaseq_data, scale="row", clustering_distance_rows="correlation", clustering_distance_cols = "correlation", show_rownames =F, show_colnames =T, cluster_cols=F, fontsize_row=4, main=paste(Treatment_rnaseq,"-",timept,"-DEGenes",sep="" ))


#T14
timept=14
inputFILE=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/deShrink.group",timept,Treatment_rnaseq,"-group",timept,"RN.txt",sep="")

input_data=read.delim(inputFILE,header=T)

if(timept==3 || timept==7)
{
timept2=paste("0",timept,sep="")
} else
{
timept2=timept       
}        
        
#Get rlog values #
ctrl_cols=c(paste("RN",timept2,"01",sep=""),paste("RN",timept2,"02",sep=""),paste("RN",timept2,"03",sep=""))
trt_cols=c(paste(Treatment_rnaseq,timept2,"01",sep=""),paste(Treatment_rnaseq,timept2,"02",sep=""),paste(Treatment_rnaseq,timept2,"03",sep=""))
subset_rnaseq_data=subset(rld,rownames(rld) %in% rownames(input_data),select=c(ctrl_cols,trt_cols))

pheatmap(subset_rnaseq_data, scale="row", clustering_distance_rows="correlation", clustering_distance_cols = "correlation", show_rownames =F, show_colnames =T, cluster_cols=F, fontsize_row=4, main=paste(Treatment_rnaseq,"-",timept,"-DEGenes",sep="" ))


#T21
timept=21
inputFILE=paste("/users/ssdelci/ddatta/Projects/Ritobrata_Ghose/rnaseq/SDELCISAR_13/STAR/DESeq2/deShrink.group",timept,Treatment_rnaseq,"-group",timept,"RN.txt",sep="")

input_data=read.delim(inputFILE,header=T)

if(timept==3 || timept==7)
{
timept2=paste("0",timept,sep="")
} else
{
timept2=timept       
}        
        
#Get rlog values #
ctrl_cols=c(paste("RN",timept2,"01",sep=""),paste("RN",timept2,"02",sep=""),paste("RN",timept2,"03",sep=""))
trt_cols=c(paste(Treatment_rnaseq,timept2,"01",sep=""),paste(Treatment_rnaseq,timept2,"02",sep=""),paste(Treatment_rnaseq,timept2,"03",sep=""))
subset_rnaseq_data=subset(rld,rownames(rld) %in% rownames(input_data),select=c(ctrl_cols,trt_cols))

pheatmap(subset_rnaseq_data, scale="row", clustering_distance_rows="correlation", clustering_distance_cols = "correlation", show_rownames =F, show_colnames =T, cluster_cols=F, fontsize_row=4, main=paste(Treatment_rnaseq,"-",timept,"-DEGenes",sep="" ))

```



```{r Plots, Treatment R1 vs. RN , warning=FALSE, echo=FALSE}

timept=3
treatname=paste("group",timept,"R1",sep="")  
ctname=paste("group",timept,"RN",sep="") 

dotplot(egoUP_3R1, showCategory=20, font.size=8, title=paste("Gene Ontology, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

dotplot(egoDOWN_3R1, showCategory=20, font.size=8, title=paste("Gene Ontology, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))


if(nrow(xx_3R1)>0)
{        
barplot(xx_3R1,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}
if(nrow(yy_3R1)>0)
{        
barplot(yy_3R1,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}

timept=7
treatname=paste("group",timept,"R1",sep="")  
ctname=paste("group",timept,"RN",sep="") 

dotplot(egoUP_7R1, showCategory=20, font.size=8, title=paste("Gene Ontology, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

dotplot(egoDOWN_7R1, showCategory=20, font.size=8, title=paste("Gene Ontology, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

if(nrow(xx_7R1)>0)
{        
barplot(xx_7R1,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}
if(nrow(yy_7R1)>0)
{        
barplot(yy_7R1,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}


timept=14
treatname=paste("group",timept,"R1",sep="")  
ctname=paste("group",timept,"RN",sep="") 

dotplot(egoUP_14R1, showCategory=20, font.size=8, title=paste("Gene Ontology, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

dotplot(egoDOWN_14R1, showCategory=20, font.size=8, title=paste("Gene Ontology, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))


if(nrow(xx_14R1)>0)
{        
barplot(xx_14R1,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}
if(nrow(yy_14R1)>0)
{        
barplot(yy_14R1,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}




timept=21
treatname=paste("group",timept,"R1",sep="")  
ctname=paste("group",timept,"RN",sep="") 

dotplot(egoUP_21R1, showCategory=20, font.size=8, title=paste("Gene Ontology, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))

dotplot(egoDOWN_21R1, showCategory=20, font.size=8, title=paste("Gene Ontology, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))


if(nrow(xx_21R1)>0)
{        
barplot(xx_21R1,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, UP", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}
if(nrow(yy_21R1)>0)
{        
barplot(yy_21R1,showCategory = 20,font.size=8) + ggtitle(paste("Reactome Pathways, DOWN", treatname, "-", ctname,sep=" ")) +  theme(plot.title = element_text(hjust = 0.5))
}

```




```{r Plots, echo=FALSE}
## Number of DEG ##

##Only rnaseq data##

Diff_Regions=c(5372,3810,2787,1093,4552,4037,2758,3222,3720,2358,2274,845,4986,3299,2612,795,3981,3440,2466,2816,3539,2296,2384,799)
dat1=data.frame( Treatment=factor(rep(rep(c("RK", "R6", "R1"),each=4),2)), Direction=factor(rep(c("UP","DOWN"),each=12)), Timept=factor(rep(c(3,7,14,21),6)), Diff_Regions=Diff_Regions-1)
dat1$Treatment=factor(dat1$Treatment,levels = c("RK", "R6", "R1"),ordered=TRUE)
dat1$Direction=factor(dat1$Direction,levels = c("UP", "DOWN"),ordered=TRUE)
dat1[which(dat1$Direction=="DOWN"),4]=-dat1[which(dat1$Direction=="DOWN"),4]

ggplot(dat1, aes(fill=Direction, y=Diff_Regions, x=Timept)) + geom_bar(position="identity", stat="identity",width=0.5) + xlab("") + ylab("Differentially Expressed Genes") + ggtitle("RNAseq") + scale_fill_manual(values=c("#CC0033","#0099CC")) + theme_bw() + theme(plot.title = element_text(hjust = 0.5)) + facet_wrap(~ Treatment, ncol=2)


# ##Combining with atac-seq data##
# 
# Diff_Regions=c(686,176,57,31,125,1681,203,330,433,42,11,15,1518,134,44,8,89,524,145,465,829,106,65,3)
# dat1=data.frame( Treatment=factor(rep(rep(c("RK", "R6", "R1"),each=4),2)), Direction=factor(rep(c("UP","DOWN"),each=12)), Timept=factor(rep(c(3,7,14,21),6)), Diff_Regions=Diff_Regions-1)
# dat1$Treatment=factor(dat1$Treatment,levels = c("RK", "R6", "R1"),ordered=TRUE)
# dat1$Direction=factor(dat1$Direction,levels = c("UP", "DOWN"),ordered=TRUE)
# dat1[which(dat1$Direction=="DOWN"),4]=-dat1[which(dat1$Direction=="DOWN"),4]
# 
# 
# ggplot(dat1, aes(fill=Direction, y=Diff_Regions, x=Timept)) + geom_bar(position="identity", stat="identity",width=0.5) + xlab("") + ylab("Differentially Expressed Genes") + ggtitle("RNAseq - ATACseq") + scale_fill_manual(values=c("#CC0033","#0099CC")) + theme_bw() + theme(plot.title = element_text(hjust = 0.5)) + facet_wrap(~ Treatment, ncol=2)
# 

```
